exp/controllers/machinepool_controller_phases.go

```diff
@@ -60,11 +60,21 @@ func (r *MachinePoolReconciler) reconcilePhase(mp *expv1.MachinePool) {
 		mp.Status.SetTypedPhase(expv1.MachinePoolPhaseProvisioned)
 	}
 
-	// Set the phase to "running" if there is a NodeRef field.
-	if mp.Status.InfrastructureReady && len(mp.Status.NodeRefs) == int(mp.Status.ReadyReplicas) {
+	// Set the phase to "running" if the number of ready replicas is equal to desired replicas.
+	if mp.Status.InfrastructureReady && *mp.Spec.Replicas == mp.Status.ReadyReplicas {
 		mp.Status.SetTypedPhase(expv1.MachinePoolPhaseRunning)
 	}
 
+	// Set the phase to "scalingUp" if the infrastructure is scaling up.
+	if mp.Status.InfrastructureReady && *mp.Spec.Replicas > mp.Status.ReadyReplicas {
+		mp.Status.SetTypedPhase(expv1.MachinePoolPhaseScalingUp)
+	}
+
+	// Set the phase to "scalingDown" if the infrastructure is scaling down.
+	if mp.Status.InfrastructureReady && *mp.Spec.Replicas < mp.Status.ReadyReplicas {
```

Comment:
I'm not entirely sure if Status.ReadyReplicas is the best way to describe "current" replicas (Spec.Replicas being desired), the Machine Deployment phases do something different with unavailable replicas:

```
	// This is the same as unavailableReplicas, but we have to recalculate because unavailableReplicas
	// would have been reset to zero above if it was negative
	if totalReplicas-availableReplicas < 0 {
		status.Phase = string(clusterv1.MachineDeploymentPhaseScalingDown)
```
https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machinedeployment_sync.go#L382-L383